btree/btree.c

```diff
@@ -2377,6 +2411,288 @@ void m0_btree_ut_node_add_del_rec(void)
+	m0_be_tx_prep(tx, NULL);
+
+	/**
+	 *  Run valid scenario:
+	 *  1) Create a btree
+	 *  2) Adds a few records to the created tree.
+	 *  3) Confirms the records are present in the tree.
+	 *  4) Deletes all the records from the tree.
+	 *  4) Close the btree
+	 *  5) Destroy the btree
+	 */
+
+	/** Create temp node space and use it as root node for btree */
+	temp_node = m0_alloc_aligned((1024 + sizeof(struct nd)), 10);
+	M0_BTREE_OP_SYNC_WITH(&b_op.bo_op,
+			      m0_btree_create(temp_node, 1024, &btree_type,
+					      tx, &b_op));
+
+	for (i = 0; i < 2048; i++) {
+		uint64_t             key;
+		uint64_t             value;
+		struct cb_data       put_data;
+		struct m0_btree_key  put_key;
+		struct m0_bufvec     put_value;
+		m0_bcount_t          ksize  = sizeof key;
+		m0_bcount_t          vsize  = sizeof value;
+		void                *k_ptr  = &key;
+		void                *v_ptr  = &value;
+
+		/**
+		 *  There is a very low possibility of hitting the same key
+		 *  again. This is fine as it helps debug the code when insert
+		 *  is called with the same key instead of update function.
+		 */
+		key = value = random();
+
+		if (!first_key_initialized) {
+			first_key = key;
+			first_key_initialized = true;
+		}
+
+		put_key.k_data     = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize);
+		put_value          = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize);
+
+		put_data.key       = &put_key;
+		put_data.value     = &put_value;
+
+		ut_cb.c_act        = btree_kv_put_cb;
+		ut_cb.c_datum      = &put_data;
+
+		M0_BTREE_OP_SYNC_WITH(&kv_op.bo_op,
+				      m0_btree_put(b_op.bo_arbor, tx,
+						   &put_key, &ut_cb, 0,
+						   &kv_op));
+	}
+
+	{
+		uint64_t             key;
+		uint64_t             value;
+		struct cb_data       get_data;
+		struct m0_btree_key  get_key;
+		struct m0_bufvec     get_value;
+		m0_bcount_t          ksize     = sizeof key;
+		m0_bcount_t          vsize     = sizeof value;
+		void                *k_ptr    = &key;
+		void                *v_ptr    = &value;
+		uint64_t             search_key;
+		void                *search_key_ptr = &search_key;
+		m0_bcount_t          search_key_size = sizeof search_key;
+		struct m0_btree_key  search_key_in_tree;
+
+		get_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize);
+		get_value      = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize);
+
+		get_data.key    = &get_key;
+		get_data.value  = &get_value;
+
+		ut_cb.c_act   = btree_kv_get_cb;
+		ut_cb.c_datum = &get_data;
+
+		search_key = first_key;
+
+		search_key_in_tree.k_data =
+			M0_BUFVEC_INIT_BUF(&search_key_ptr, &search_key_size);
+
+		M0_BTREE_OP_SYNC_WITH(&kv_op.bo_op,
+				      m0_btree_get(b_op.bo_arbor,
+						   &search_key_in_tree,
+						   &ut_cb, 0, &kv_op));
+
+		for (i = 1; i < 2048; i++) {
+			search_key = key;
+			M0_BTREE_OP_SYNC_WITH(&kv_op.bo_op,
+					      m0_btree_nxt(b_op.bo_arbor,
+							   &search_key_in_tree,
+							   &ut_cb, 0, &kv_op));
+		}
+	}
+
+	m0_btree_close(b_op.bo_arbor);
```

Comment:
Will `m0_btree_close` free up the btree (temp_node) or `m0_btree_destroy` should be called to free up memory?

Reply:
You are right we should call destroy at the end. I will make the changes.